src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -95,28 +96,47 @@ private void OnProjectsChanged(ConnectionInformation connection, ProjectInformat
 
         #region Workflow steps
 
+        private static void AbortWorkflow(IProgressController controller, CancellationToken token)
+        {
+            bool aborted = controller.TryAbort();
+            Debug.Assert(aborted || token.IsCancellationRequested, "Failed to abort the workflow");
+        }
+
         internal /* for testing purposes */ void ConnectionStep(IProgressController controller, CancellationToken cancellationToken, ConnectionInformation connection, IProgressStepExecutionEvents notifications)
         {
             this.owner.UserNotification?.HideNotification(NotificationIds.FailedToConnectId);
+            this.owner.UserNotification?.HideNotification(NotificationIds.BadServerPluginId);
 
             notifications.ProgressChanged(connection.ServerUri.ToString(), double.NaN);
 
             ProjectInformation[] projects = this.owner.SonarQubeService.Connect(connection, cancellationToken)?.ToArray();
```

Comment:
This highlights my current gripe which is we've fused the downloading of the project information with the establishment of the server connection. I had started down this approach already, so might be better to go back to that approach. I had stopped because of too much churn.